public/app/features/dashboard-scene/settings/AnnotationsEditView.tsx

```diff
@@ -51,11 +51,32 @@ export class AnnotationsEditView extends SceneObjectBase<AnnotationsEditViewStat
     return this._dashboard;
   }
 
-  public onNew = () => {
+  // Get the datasource that supports annotations
+  public getDataSourceRefForAnnotation = async () => {
+    // get current default datasource ref from instance settings
+    const defaultInstanceDS = getDataSourceSrv().getInstanceSettings(null);
+    // get datasource ref from default datasource
+    const defaultDSRef = getDataSourceRef(defaultInstanceDS!);
+    const dataSourceSrv = await getDataSourceSrv().get(defaultDSRef);
+
+    // if default ds does not support annotations, return the grafana datasource
+    if (!dataSourceSrv?.annotations) {
+      const dsGrafana = getDataSourceSrv().getInstanceSettings('-- Grafana --');
+      if (!dsGrafana) {
+        throw new Error('Grafana datasource not found');
+      }
+      const grafanaDataSourceRef = getDataSourceRef(dsGrafana);
+      return grafanaDataSourceRef;
+    } else {
+      return defaultDSRef;
+    }
```

Comment:
there is an annotations flag in plugin json if the datasource supports annotations so you should be able to check that. 

that flag should be on the defaultInstanceDS (InstanceSettings) 

that is better than checking dataSourceSrv.annotations as there are older data sources that implement it via the old annotationQuery function that is directly on the DataSource class